### PR TITLE
[WIP] Fix after-and-docked in let bindings

### LINKED ISSUE #967: Bug: break-separators=after-and-docked doesn't affect type definitions
```
type t =
  { hash: hash_type;
    file_mode: mode option;
    funs: id list;
    classes: id list;
    typedefs: id list;
    consts: id list;
    comments: (Pos.t * comment) list option }

let empty_t =
  { hash= None;
    file_mode= None;
    funs= [];
    classes= [];
    typedefs= [];
    consts= [];
    comments= Some [] }
```

Produces:

```
$ ocamlformat --break-separators=after-and-docked demo1.ml
type t = {
  hash: hash_type;
  file_mode: mode option;
  funs: id list;
  classes: id list;
  typedefs: id list;
  consts: id list;
  comments: (Pos.t * comment) list option;
}

let empty_t =
  {
    hash= None;
    file_mode= None;
    funs= [];
    classes= [];
    typedefs= [];
    consts= [];
    comments= Some [];
  }
```

This is inconsistent with the other values of `break-separators`, which do affect type definitions.

### PR DESCRIPTION
**Pull request by gpetiot**
Fix #967 
not ready to be reviewed.

### CODE REVIEW

File: test/passing/break_cases-align.ml.ref [hunk 1]
```diff
@@ -25,8 +25,7 @@ let f =
      |M ->
     match y with
     | O -> 5
-    | P when h x -> (
-        function
+    | P when h x -> ( function
```
Comment (gpetiot) [inline test/passing/break_cases-align.ml.ref]: This kind of change is unavoidable with this new design

Comment (Julow) [reply test/passing/break_cases-align.ml.ref]: Why ? This is not a collection expression.

**Comment by gpetiot** [comment]
Too many collateral damage reported by test_branch, I will do it in smaller steps in smaller PRs without having the same behavior for collection expressions and functions (although both can be split (pre_body/body).

**Comment by gpetiot** [comment]
Replaced by #1022

**Pull request closed without merging**